lib/chef/provider/service/insserv.rb

```diff
@@ -36,7 +36,9 @@ def load_current_resource
           super
 
           # Look for a /etc/rc.*/SnnSERVICE link to signify that the service would be started in a runlevel
-          if Dir.glob("/etc/rc**/S*#{Chef::Util::PathHelper.escape_glob_dir(current_resource.service_name)}").empty?
+          service_name = Chef::Util::PathHelper.escape_glob_dir(current_resource.service_name)
+
+          if Dir.glob("/etc/rc**/**/S*#{service_name}").empty?
```

Comment:
This should probably be `"/etc/rc*/**/S*#{service_name}"`

I have concerns that if you changed this to `"/etc/rc*/*/S*#{service_name}"` and updated the unit tests to have the same stub, that the unit tests would pass, the tests would pass on sles11 and then the code would actually fail on rhel (which is emphasis on the fact that we need func tests in rhel around this)

Reply:
Updated to `/etc/rc*/**/S*#{service_name}`.